zkevm-circuits/src/evm_circuit/op_execution/utils/math_gadgets.rs

```diff
@@ -0,0 +1,85 @@
+use super::super::CaseAllocation;
+use super::super::Cell;
+use crate::util::Expr;
+use halo2::plonk::Error;
+use halo2::{arithmetic::FieldExt, circuit::Region, plonk::Expression};
+
+use super::constraint_builder::ConstraintBuilder;
+
+// a == 0
```

Comment:
I was confused at first for not being aware of its returning value. Suggest adding the doc to emphasize the return value part.
Notes to other reviewers: This gadget is the equivalent of circomlib's `IsZero` templates https://github.com/iden3/circomlib/blob/master/circuits/comparators.circom#L23
```suggestion
// a == 0
// The gadget returns 1 if the input value `a` is 0, and it returns 0 otherwise. 
```